src/main/java/org/owasp/benchmark/score/WriteTime.java

```diff
@@ -0,0 +1,389 @@
+				e.toString();
+			}
+		}
+		return "";
+	}
+
+	public static void listPathFiles(String path) {
+		String files;
+		File folder = new File(path);
+		File[] listOfFiles = folder.listFiles();
+
+		for (int i = 0; i < listOfFiles.length; i++) {
+
+			if (listOfFiles[i].isFile()) {
+				files = listOfFiles[i].getName();
+				System.out.println(files);
+			}
+		}
+	}
+
+	public void deletePreviousResults(String toolName, String toolVersion,
+			String benchmarkVersion) {
+		if (!toolName.equals("")) {
+			File targetDir = new File("results/");
+			if (targetDir.exists()) {
+				File[] files = targetDir.listFiles();
+				for (File file : files) {
+					if (file.isFile()
+							&& (file.getName().startsWith(
+									"Benchmark_" + benchmarkVersion + "-"
+											+ toolName + "-v" + toolVersion) && (file
+									.getName().endsWith("xml")))) {
+						file.delete();
+					}
+				}
+				System.out.println("Deleted previously generated results files.");
+			}
+		}
+	}
+
+	public void resultsFileName(String tool, String benchmarkVersion,
+			String times, String toolVersion) {
+		String name = "results/Benchmark_" + benchmarkVersion + "-" + tool
+				+ "-v" + toolVersion + "-" + times + ".xml";
+		File file = null;
+
+		File targetDir = new File("results/");
+		if (!targetDir.exists()) {
+			targetDir.mkdir();
+		}
+
+		// System.out.println("inside results file: "+tool);
+		switch (tool) {
+			case "findbugs":
+				file = new File(FINDBUGS_FILE);
+				if (file.exists()) {
+					file.renameTo(new File(name));
+				}
+				break;
+			case "pmd":
+				file = new File(PMD_FILE);
+				if (file.exists()) {
+					file.renameTo(new File(name));
+				}
+				break;
+			case "sonar":
+				file = new File(SONAR_FILE);
+				if (file.exists()) {
+					file.renameTo(new File(name));
+				}
+				break;
+			case "findsecbugs":  // findsecbugs now runs with spotbugs, not findbugs
+			case "spotbugs":
+				file = new File(SPOTBUGS_FILE);
+				if (file.exists()) {
+					file.renameTo(new File(name));
+				}
+				break;
+			case "crawler":
+				file = new File("results/" + getFindFile("results", CONTRAST_FILE + benchmarkVersion + "-Contrast"));
+				if (file.exists()) {
+					file.renameTo(new File("results/" + file.getName().replace(".zip", "-" + times + ".zip")));
+				}
+				break;
+		}
+	}
+
+	public void writeSonarResults() {
+
+		int page = 1;
+		int total = 1;
+		JSONArray issues = new JSONArray();
+		JSONObject json = null;
+
+		try {
+			while (issues.length() < total && page <= 20) {
+				json = new JSONObject(getSonarResults(SONAR_URL + "/api/issues/search?resolved=false&ps=500&p=" + page));
+				total = (int) json.get("total");
+
+				JSONArray issueSubset = json.getJSONArray("issues");
+				for (int i = 0; i < issueSubset.length(); i++) {
+					issues.put(issueSubset.get(i));
+				}
+				page++;
+			}
+
+			json.put("issues", issues);
+
+			String xml = XML.toString(json);
+			java.io.FileWriter fw = new java.io.FileWriter(SONAR_FILE);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fw*. The resource is referenced in statements at the following line: 304. The resource closure statement is at line: 305. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by Writer.write. Either a) close *fw* in a try-finally block or b) close the resource by declaring *fw* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).